cookbooks/bcpc-hadoop/templates/default/zk_zookeeper-env.sh.erb

```diff
@@ -32,5 +36,16 @@ SERVER_JVMFLAGS="$SERVER_JVMFLAGS -Djava.security.auth.login.config=/etc/zookeep
 CLIENT_JVMFLAGS="$CLIENT_JVMFLAGS -Djava.security.auth.login.config=/etc/zookeeper/conf/zookeeper-client.jaas"
 <% end %>
 
+<%
+  auto_size = (node['memory']['total'].to_i *
+               node['bcpc']['hadoop']['zookeeper']['xmx']['max_ratio']/1024)
+  heap = [node['bcpc']['hadoop']['zookeeper']['xmx']['max_size'],
+          auto_size.floor].min
+  newsize = [(0.125*heap).ceil, 3072].min
+-%>
+SERVER_JVMFLAGS="$SERVER_JVMFLAGS -Xms<%= heap %>m -Xmx<%= heap %>m " \
+                "-Xmn<%= newsize %>m " \
+                "<%= node['bcpc']['hadoop']['zookeeper']['gc_opts'] %>"
```

Comment:
FC0035 is deprecated but I think this template has enough access to various parts of the `node` Object that it is worth enforcing.

See https://github.com/Foodcritic/foodcritic/issues/62 for background.  Enforcing it is getting too draconian but it still makes sense to enforce in the other extreme.

Reply:
@aespinosa I am not quite sure what direction you are wanting this to go in but at this time a change for node object access here should be consistent with the other -env.sh files we have and would be a broader scope of work than I am comfortable with for this change. For reference, we have the following if you want to file an issue to tackle this?
* [flume-env.sh](https://github.com/bloomberg/chef-bach/blob/f998f3edeee54fbed9f7dfad27be1bb44a23be80/cookbooks/bcpc-hadoop/templates/default/flume_flume-env.sh.erb)
* [hadoop-env.sh](https://github.com/bloomberg/chef-bach/blob/5a69ccdd1de88b22e29309b9df4ffc6ae6e6c17e/cookbooks/bcpc-hadoop/templates/default/hdp_hadoop-env.sh.erb)
* [mapred-env.sh](https://github.com/bloomberg/chef-bach/blob/491bb6779b765cf25267dcd008008f6ba0e8e0d5/cookbooks/bcpc-hadoop/templates/default/hdp_mapred-env.sh.erb)
* [httpfs-env.sh](https://github.com/bloomberg/chef-bach/blob/aa6b333717cb203c0ced7ae3c867372a955ef701/cookbooks/bcpc-hadoop/templates/default/httpfs-env.sh.erb)
* [oozie-env.sh](https://github.com/bloomberg/chef-bach/blob/4de444b558684b0b2e7e59a65ace77035ea912b3/cookbooks/bcpc-hadoop/templates/default/ooz_oozie-env.sh.erb)
* https://github.com/bloomberg/chef-bach/blob/aa6b333717cb203c0ced7ae3c867372a955ef701/cookbooks/bcpc-hadoop/templates/default/sq_sqoop-env.sh.erb